File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    st1 = [1, 2, 3, 4, 5]
    st2 =[2, 1, 2, 3, 2, 4, 2, 5]
    st3 =[3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    answer = []
    correct = [0,0,0]
    
    for i in range(len(answers)):
        if answers[i] == st1[i%5]:
            correct[0] += 1
        if answers[i] == st2[i%8]:
            correct[1] += 1
        if answers[i] == st3[i%10]:
            correct[2] += 1
            
    winner = max(correct)
    for i in range(len(correct)):
        if correct[i] == winner:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] pattern1 = {1, 2, 3, 4, 5}; // 길이 5
        int[] pattern2 = {2, 1, 2, 3, 2, 4, 2, 5}; // 길이 8
        int[] pattern3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}; // 길이 10
        
        int[] scores = {0,0,0};
        
        for (int i = 0; i < answers.length; i++) {
            if (answers[i] == pattern1[i % pattern1.length]) scores[0]++;
            if (answers[i] == pattern2[i % pattern2.length]) scores[1]++;
            if (answers[i] == pattern3[i % pattern3.length]) scores[2]++;
        }
        
        int maxScore =Math.max(scores[0], Math.max(scores[1],scores[2]));
        
        List<Integer> list = new ArrayList<>();
        
        for(int i=0; i < scores.length; i++){
            if (scores[i] == maxScore){
                list.add(i+1);
            }
        }
        int[] result = new int[list.size()];
        for(int i= 0; i<list.size(); i++){
            result[i] = list.get(i);
        }
        return result;
    }
}